include/seqan3/alignment/configuration/align_config_debug.hpp

```diff
@@ -0,0 +1,46 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::align_cfg::debug.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/alignment/configuration/detail.hpp>
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
+
+namespace seqan3::align_cfg
+{
+
+//!\cond DEV
+//!\brief The configuration for debugging alignment algorithms.
+//!\ingroup alignment_configuration
+struct debug_type : public pipeable_config_element<debug_type, bool>
+{
+    //!\privatesection
+    //!\brief Internal id to check for consistent configuration settings.
+    static constexpr detail::align_config_id id{detail::align_config_id::debug};
+};
+
+/*!\brief Configuration element for debugging the alignment algorithm.
+ * \ingroup alignment_configuration
+ *
+ * \details
+ *
+ * Using this configuration allows to output the alignment matrices from the DP algorithm using the
+ * returned seqan3::alignment_result.
+ * The score matrix is always accessible, while the trace matrix can only be computed if an alignment was
+ * requested via the seqan3::align_cfg::result configuration.
+ *
+ * \note This configuration is only useful for debugging purposes as it can have a significant impact on the
+ *       performance.
+ */
+inline constexpr debug_type debug{};
+//!\endcond
+}  // namespace seqan3::align_cfg
```

Comment:
why not ::detail?

Reply:
Mhmm, yes, I could put the config in detail.

Reply:
For now it is only visible for developer but I found it nicer to have the chain use `align_cfg` as well. But this is not so important and detail would be fine as well

Reply:
So I changed it to some global config thing. But note that the actual debug thing need to be in align_cfg to disambiguate it from other definitions in other namespaces. Like for the search you need your own config element since it uses internal enum ids to validate the combinability. Thus leaving the actual variable in align_cfg makes sense. But I put the global type into detail and to some generic place usable for other algorithms

Reply:
ok